builders/relayedTxV2Builder.go

```diff
@@ -86,7 +90,7 @@ func (rtb *relayedTxV2Builder) Build() (*transaction.FrontendTransaction, error)
 
```

Comment:
can we also add checks for the following fields to match between the inner and the relayed transactions?
```
GasPrice
ChainID
Version
Options
```